### PR TITLE
[GAE-Java] Speed up Java App runtime codegen& compiling

### LINKED ISSUE #2149: [Performance] Running Java app takes too long
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

It takes too long to running java app on GAE, we may blame the java codegen procedure.

### PR DESCRIPTION
**Pull request by zhanglei1949**
## Related issue number

Fixes #2149

### CODE REVIEW

File: analytical_engine/java/grape-annotation/src/main/java/com/alibaba/graphscope/annotation/GrapeSkipAnnotationProcessor.java [hunk 1]
```diff
@@ -194,7 +242,7 @@ private void check(String[] types) {
 
     private void writeTypeSpec(String packageName, TypeSpec typeSpec, TypeElement typeElement) {
         JavaFile javaFile = JavaFile.builder(packageName, typeSpec).build();
-        messager.printMessage(Kind.NOTE, javaFile.toString(), typeElement);
+        //        messager.printMessage(Kind.NOTE, javaFile.toString(), typeElement);
```
Comment (sighingnow) [inline analytical_engine/java/grape-annotation/src/main/java/com/alibaba/graphscope/annotation/GrapeSkipAnnotationProcessor.java:245]: Delete if not used.

File: analytical_engine/java/grape-runtime/pom.xml [hunk 2]
```diff
@@ -64,14 +64,18 @@
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
     </dependency>
-    <dependency>
+      <dependency>
```
Comment (sighingnow) [inline analytical_engine/java/grape-runtime/pom.xml:67]: Seems that the indention of newly added blocks is not aligned with existing dependencies..

Comment (sighingnow) [reply analytical_engine/java/grape-runtime/pom.xml:67]: BTW I have noticed that we have switch forth and back the logger for many times. Is there any consideration for that?

File: coordinator/gscoordinator/utils.py [hunk 3]
```diff
@@ -227,6 +228,39 @@ def get_graph_sha256(attr):
...
+):
+    splited = cpp_graph_type.split("<")
+    java_app_type_params = java_class_template_str[:-1].split("<")[-1].split(",")
+    num_type_params = 0
+    if len(splited) != 2:
+        raise Exception("Unrecoginizable graph template str: {}".format(cpp_graph_type))
+    if splited[0] == "vineyard::ArrowFragment":
+        if app_class.find("Property") == -1:
+            logger.error("Expected property app")
+            return False
```
Comment (sighingnow) [inline coordinator/gscoordinator/utils.py:242]: I see no reason why you don't add the error message into the exception message (as you throw exception if `check_java_app_graph_consistency ` fails)...

File: python/graphscope/tests/unittest/test_java_app.py [hunk 4]
```diff
@@ -134,8 +134,11 @@ def test_sssp_property_vertex_data(
     p2p_project_directed_graph,
     projected_graph_sssp_class,
 ):
-    sssp = JavaApp(full_jar_path=demo_jar, java_app_class=projected_graph_sssp_class)
-    sssp(p2p_project_directed_graph, src=6, threadNum=1)
+    graphscope_session.add_lib(demo_jar)
+    sssp = load_app(algo="java_pie:{}".format(projected_graph_sssp_class))
+    sssp(p2p_project_directed_graph._project_to_simple(), src=6, threadNum=1)
+    # sssp = JavaApp(full_jar_path=demo_jar, java_app_class=projected_graph_sssp_class)
+    # sssp(p2p_project_directed_graph, src=6, threadNum=1)
```
Comment (sighingnow) [inline python/graphscope/tests/unittest/test_java_app.py:141]: Delete them if you won't need.

**Review by sighingnow — APPROVED** [review]

**Comment by sighingnow** [comment]
How much speed up of building java apps can we expect after this pull request?

File: analytical_engine/java/grape-runtime/pom.xml (hunk 2 above)
Comment (zhanglei1949) [reply analytical_engine/java/grape-runtime/pom.xml:67]: Actually in previous PR, the logging problem has not been fixed(Only `ERROR` log is visible), in this PR, we replace log4j.properties with log4j.xml and fix this issue.

**Comment by zhanglei1949** [comment]
> How much speed up of building java apps can we expect after this pull request?

About 2 times speeds up. With this PR merged , we can reduce the cost time of Java Python test from 25-30min to around 10min.

**Pull request merged**